### PR TITLE
Flesh out YAML API documentation

### PR DESCRIPTION
**Pull request by kdmccormick**
@edx/masters-neem 

**How to test**
* `git checkout kdmccormick/api-docs`
* `docker restart edx.devstack.registrar`
* Navigate to http://localhost:18734/login, log in as edx@example.com so all endpoints are visible
* Navigate to http://localhost:18734/api-docs and make sure the look right and are easy to understand

### CODE REVIEW

File: .api-generated.yaml [hunk 1]
```diff
@@ -19,23 +19,132 @@ program_path_param:
...
+      $ref: '#/models/EnrollmentResult'
+    examples:
+      application/json:
+        student_0128fe4a: invalid-status
+        student_aae45c81: not-found
+
+student_key:
+  type: string
+  description: >
+    key that uniquely identifies student within organization.
```
Comment (reillz10) [inline .api-generated.yaml]: Need a capital `k` to start the sentence

File: .api-generated.yaml [hunk 2]
```diff
@@ -537,54 +1015,109 @@ models:
...
     type: array
     items:
       type: object
       properties:
         student_key:
           type: string
-        email:
-          type: string
+          description: >
+            key that uniquely identifies student within organization.
```
Comment (reillz10) [inline .api-generated.yaml]: Same

File: api.yaml [hunk 3]
```diff
@@ -20,23 +20,132 @@ program_path_param: &program_path_param
...
+      $ref: '#/models/EnrollmentResult'
+    examples:
+      application/json:
+        student_0128fe4a: invalid-status
+        student_aae45c81: not-found
+
+student_key: &student_key
+  type: string
+  description: >
+    key that uniquely identifies student within organization.
```
Comment (reillz10) [inline api.yaml]: same

**Review by reillz10 — CHANGES_REQUESTED** [review]
From what I can see, just a couple of "k"'s that need to be capitalized. Other than that looks good!

File: api.yaml (hunk 3 above)
Comment (kdmccormick) [reply api.yaml]: Good call; ready for another look!

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**